Maintenance: CSS flex guide: updates to guide 5

estelle opened the pull request:
Flexbox is well-supported, and the content value is fully supported. This update freshens up the guide and removes a lot of words to tighten it up, then adds clarification text, so it's about the same size :(.

updating guide as part of CSS baseline / interop https://github.com/openwebdocs/project/issues/189 - subcomponent https://github.com/openwebdocs/project/issues/202

estelle committed:
CSS maintenance: flex guide 5

estelle committed:
Merge branch 'main' into flex-guide5

estelle committed:
Update files/en-us/web/css/css_flexible_box_layout/controlling_ratios_of_flex_items_along_the_main_axis/index.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

dipikabh commented on files/en-us/web/css/css_flexible_box_layout/controlling_ratios_of_flex_items_along_the_main_axis/index.md:
```diff
@@ -16,164 +16,156 @@ Our three properties control the following aspects of a flex item's flexibility:
+In the same way as `flex-grow`, you can give flex-items different `flex-shrink` factors. This can help change the default behavior if, for example, you want an item to shrink more or less rapidly than its siblings or to not shrink at all.
 
-In the following live example the first item has a `flex-shrink` factor of 1, the second `0` (so it won't shrink at all), and the third `4`. The third item therefore shrinks more rapidly than the first. Play around with the different values — as for `flex-grow` you can use decimals or larger numbers here. Choose whatever makes most sense to you.
+In this example, the first item has a `flex-shrink` factor of 1, the second `0` (so it won't shrink at all), and the third `4`, for a total of 5 shrink factors. The third item therefore shrinks approximately four times more rapidly than the first, but neither will shrink below their `min-content` width. Play around with the different values — as with `flex-grow`, you can use decimals or larger nu ...
```
```suggestion
In this example, the first item has a `flex-shrink` factor of `1`, the second item `0` (so it won't shrink at all), and the third item `4`, making a total of `5` shrink factors. The third item, therefore, shrinks approximately four times more rapidly than the first, but neither will shrink below their `min-content` width. Play around with the different values: as with `flex-grow`, you can use decimals or larger numbers here as well.
```

dipikabh commented on files/en-us/web/css/css_flexible_box_layout/controlling_ratios_of_flex_items_along_the_main_axis/index.md:
```diff
@@ -16,164 +16,156 @@ Our three properties control the following aspects of a flex item's flexibility:
 ## Mastering sizing of flex items
 
-The key to really understanding how flex item sizing works is in understanding the number of things that come into play. Consider the following aspects, which we have already discussed in these guides:
+To really understand how flex item sizing works, you need to understand all these factors that come into play. Consider the following aspects, which we have already discussed in these guides:
```
these two can be combined:
```suggestion
To understand how flex item sizing works, you need to consider the factors below, which we've discussed in these guides:
```

dipikabh commented on files/en-us/web/css/css_flexible_box_layout/controlling_ratios_of_flex_items_along_the_main_axis/index.md:
```diff
@@ -16,164 +16,156 @@ Our three properties control the following aspects of a flex item's flexibility:
-The key to really understanding how flex item sizing works is in understanding the number of things that come into play. Consider the following aspects, which we have already discussed in these guides:
+To really understand how flex item sizing works, you need to understand all these factors that come into play. Consider the following aspects, which we have already discussed in these guides:
 
 ### What sets the base size of the item?
```
```suggestion
### What determines the base size of an item?
```

dipikabh commented on files/en-us/web/css/css_flexible_box_layout/controlling_ratios_of_flex_items_along_the_main_axis/index.md:
```diff
@@ -16,164 +16,156 @@ Our three properties control the following aspects of a flex item's flexibility:
+3. Is `flex-basis` a length or percentage, but not zero? If so this is the size of the item (floored at `min-content`).
+4. Is `flex-basis` set to `0`? If so, the item's size is not taken into consideration for the space-sharing calculation.
 
 ### Do we have available space?
```
```suggestion
### Is there available space?
```

dipikabh commented on files/en-us/web/css/css_flexible_box_layout/controlling_ratios_of_flex_items_along_the_main_axis/index.md:
```diff
@@ -16,164 +16,156 @@ Our three properties control the following aspects of a flex item's flexibility:
+1. If we add up the widths of all the items (or heights if working in a column), is that total **less** than the total width (or height) of the container? If so, then you have positive free space and `flex-grow` comes into play.
+2. If we add up the widths of all the items (or heights if working in a column), is that total **more** than the total width (or height) of the container? If so, you have negative free space and `flex-shrink` comes into play.
 
 ### Other ways to distribute space
```
```suggestion
### What are the other ways to distribute space?
```

dipikabh commented on files/en-us/web/css/css_flexible_box_layout/controlling_ratios_of_flex_items_along_the_main_axis/index.md:
```diff
@@ -16,164 +16,156 @@ Our three properties control the following aspects of a flex item's flexibility:
 
 ### What sets the base size of the item?
 
 1. Is `flex-basis` set to `auto`, and does the item have a width set? If so, the size will be based on that width.
```
this does not need to be a numbered list IMO
```suggestion
- Is `flex-basis` set to `auto`, and does the item have a width set? If so, the size will be based on that width.
```

dipikabh commented on files/en-us/web/css/css_flexible_box_layout/controlling_ratios_of_flex_items_along_the_main_axis/index.md:
```diff
@@ -16,164 +16,156 @@ Our three properties control the following aspects of a flex item's flexibility:
-2. Is `flex-basis` set to `auto` or `content`? If so, the size is based on the item size.
-3. Is `flex-basis` a length unit, but not zero? If so this is the size of the item (floored at `min-content`).
-4. Is `flex-basis` set to `0`? if so then the item size is not taken into consideration for the space-sharing calculation.
+2. Is `flex-basis` set to `auto`? If so, the size is based on the item size.
+3. Is `flex-basis` a length or percentage, but not zero? If so this is the size of the item (floored at `min-content`).
+4. Is `flex-basis` set to `0`? If so, the item's size is not taken into consideration for the space-sharing calculation.
```
```suggestion
- Is `flex-basis` set to `auto`? If so, the size will be based on the item's content size.
- Is `flex-basis` a length or a percentage, but not zero? If so, this will be the size of the item (floored at `min-content`).
- Is `flex-basis` set to `0`? If so, the item's size will not be taken into consideration for the space-sharing calculation.
```

dipikabh commented on files/en-us/web/css/css_flexible_box_layout/controlling_ratios_of_flex_items_along_the_main_axis/index.md:
```diff
@@ -16,164 +16,156 @@ Our three properties control the following aspects of a flex item's flexibility:
 
 ### Do we have available space?
 
 Items can't grow with no positive free space, and they won't shrink unless there is negative free space.
```
```suggestion
Items can't grow without positive free space, and they won't shrink unless there is negative free space.
```

dipikabh commented on files/en-us/web/css/css_flexible_box_layout/controlling_ratios_of_flex_items_along_the_main_axis/index.md:
```diff
@@ -16,164 +16,156 @@ Our three properties control the following aspects of a flex item's flexibility:
 
-1. If we took all of the items and added up their widths (or heights if working in a column), is that total **less** than the total width (or height) of the container? If so, then you have positive free space and `flex-grow` comes into play.
-2. If we took all of the items and added up their widths (or heights if working in a column), is that total **more** than the total width (or height) of the container? If so, you have negative free space and `flex-shrink` comes into play.
+1. If we add up the widths of all the items (or heights if working in a column), is that total **less** than the total width (or height) of the container? If so, then you have positive free space and `flex-grow` comes into play.
+2. If we add up the widths of all the items (or heights if working in a column), is that total **more** than the total width (or height) of the container? If so, you have negative free space and `flex-shrink` comes into play.
```
```suggestion
- If we add up the widths of all the items (or heights if working in a column), is that total **less** than the total width (or height) of the container? If so, there will be positive free space, and `flex-grow` will come into play.
- If we add up the widths of all the items (or heights if working in a column), is that total **more** than the total width (or height) of the container? If so, there will be negative free space, and `flex-shrink` will come into play.
```

dipikabh commented on files/en-us/web/css/css_flexible_box_layout/controlling_ratios_of_flex_items_along_the_main_axis/index.md:
```diff
@@ -16,164 +16,156 @@ Our three properties control the following aspects of a flex item's flexibility:
 
 ### Other ways to distribute space
 
 If you do not want space added to the items, remember that you can deal with free space between or around items using the alignment properties described in the guide to aligning items in a flex container. The {{CSSxRef("justify-content")}} property will enable the distribution of free space between or around items. You can also use auto margins on flex items to absorb space and create gaps betwee ...
```
```suggestion
If you do not want space added to the items, remember that you can manage free space between or around items using the alignment properties described in the guide for aligning items in a flex container. The {{CSSxRef("justify-content")}} property will enable the distribution of free space between or around items. You can also use auto margins on flex items to absorb space and create gaps between items.
```

dipikabh commented on files/en-us/web/css/css_flexible_box_layout/controlling_ratios_of_flex_items_along_the_main_axis/index.md:
```diff
@@ -16,164 +16,156 @@ Our three properties control the following aspects of a flex item's flexibility:
 If you do not want space added to the items, remember that you can deal with free space between or around items using the alignment properties described in the guide to aligning items in a flex container. The {{CSSxRef("justify-content")}} property will enable the distribution of free space between or around items. You can also use auto margins on flex items to absorb space and create gaps betwee ...
 
-With all the flex tools at your disposal you will find that most tasks can be achieved, although it might take a little bit of experimentation at first.
+With all these flex properties available to you, you will find that most tasks can be achieved, although it might take a little bit of experimentation at first.
```
```suggestion
With all these flex properties available to you, you will find that most layout tasks are possible, although it might take a little bit of experimentation at first.
```

dipikabh reviewed (changes requested):
Thanks for improving this content, @estelle. I've added my editorial review.

estelle committed:
Apply suggestions from code review

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

estelle committed:
Merge branch 'main' into flex-guide5

estelle committed:
Update files/en-us/web/css/css_flexible_box_layout/controlling_ratios_of_flex_items_along_the_main_axis/index.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

estelle committed:
minor tweaks

estelle committed:
tweaks based on a re-read

estelle commented:
Thanks for the review. I've incorporated all your feedback. I also updated all the live examples because the flex container overflowed the example space on smaller screens. https://github.com/mdn/css-examples/pull/187

dipikabh reviewed (approved):
Thanks, the updates look good. There's only a very few nits. I can commit those and merge this PR.

dipikabh committed:
Apply suggestions from code review

dipikabh committed:
Update files/en-us/web/css/css_flexible_box_layout/controlling_ratios_of_flex_items_along_the_main_axis/index.md
